pydsfapi/src/pydsfapi/commands/basecommands.py

```diff
@@ -0,0 +1,362 @@
+"""
+This module contains all basic commands to be sent to the server.
+
+    Python interface to DuetSoftwareFramework
+    Copyright (C) 2020 Duet3D
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""
+from enum import IntEnum, Enum
+from typing import Optional
+
+from .codechannel import CodeChannel
+
+
+class HttpEndpointType(str, Enum):
+    """Enumeration of supported HTTP request types"""
+
+    GET = "GET"
+    POST = "POST"
+    PUT = "PUT"
+    PATCH = "PATCH"
+    TRACE = "TRACE"
+    DELETE = "DELETE"
+    OPTIONS = "OPTIONS"
+    WebSocket = "WebSocket"
+
+
+class AccessLevel(str, Enum):
+    """Defines what a user is allowed to do"""
+
+    ReadOnly = "ReadOnly"
+    ReadWrite = "ReadWrite"
+
+
+class SessionType(str, Enum):
+    """Types of user sessions"""
+
+    Local = "Local"
+    HTTP = "HTTP"
+    Telnet = "Telnet"
+
+
+class MessageType(IntEnum):
+    """Type of Resolve message"""
+
+    Success = 0
+    Warning = 1
+    Error = 2
+
+
+class BaseCommand:
+    """Base class of a command."""
+
+    @classmethod
+    def from_json(cls, data):
+        """Deserialize an instance of this class from a JSON deserialized dictionary"""
+        return cls(**data)
+
+    def __init__(self, command: str, **kwargs):
+        self.command = command
+        for key, value in kwargs.items():
+            self.__dict__[key] = value
+
+
+def acknowledge():
+    return BaseCommand("Acknowledge")
+
+
+def cancel():
+    return BaseCommand("Cancel")
+
+
+def ignore():
+    return BaseCommand("Ignore")
+
+
+def get_machine_model():
```

Comment:
Please add Deprecated comment to the user should use the `_object_` function instead. Please do so for all `_machine_` functions.

Reply:
Already deprecated them for 3.2.x? Or are you thinking now of the `3.3-dev` branch?

Reply:
Ah, I think you are looking for the comments in https://github.com/Duet3D/DSF-APIs/pull/4/files#diff-f9a619cbfb505ffc90cb728f7b7e09d502828d2f9fed1e958fcd9fc385f38d0cR245

IMO the high-level functions in the connection classes should be marked as deprecated (which they still are). The underlying `basecommands` should not really be used by users anyway. Happy to mark them as deprecated in a follow-up PR or on 3.3-dev.

Reply:
No worries, I only later realized that the functions are already marked as Deprecated.